tests/test_routes.py

```diff
@@ -159,6 +160,86 @@ def test_list_inventory_records(self):
+        test_record = InventoryFactory()
+        response = self.client.post(BASE_URL, json=test_record.serialize())
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        logging.debug('Created record, %s', Inventory().deserialize(response.get_json()))
+
+        # Get record as JSON
+        data = response.get_json()
+        logging.debug(data)
+        fields = ['quantity', 'restock_level', 'reorder_quantity']
+        quantities = [10, 15, 20]
+        restock_levels = [1, 2, 3]
+
+        """ Randomly select any 2 fields and update them """
+        # Select 1st field
+        field1 = random.choice(fields)
+        logging.debug('Field selected for update: %s', field1)
+        if field1 == 'quantity' or field1 == 'reorder_quantity':
+            logging.debug('Old value = %d', data[field1])
+            qset = set(quantities)
+            qset.remove(data[field1])
+            data[field1] = random.choice(list(qset))
+            logging.debug('New value = %d', data[field1])
+        elif field1 == 'restock_level':
+            logging.debug('Old value = %d', data[field1])
+            rlset = set(restock_levels)
+            rlset.remove(data[field1])
+            data[field1] = random.choice(list(set(rlset)))
+            logging.debug('New value = %d', data[field1])
+        elif field1 == 'active':
+            logging.debug('Old value = %r', data[field1])
+            data[field1] = not data[field1]
+            logging.debug('New value = %r', data[field1])
+
+        # Remove selected field
+        fset = set(fields)
+        fset.remove(field1)
+        fields = list(fset)
+
+        # Select another field
+        field2 = random.choice(fields)
+        if field2 == 'quantity' or field2 == 'reorder_quantity':
+            logging.debug('Old value = %d', data[field2])
+            qset = set(quantities)
+            qset.remove(data[field2])
+            data[field2] = random.choice(list(qset))
+            logging.debug('New value = %d', data[field2])
+        elif field2 == 'restock_level':
+            logging.debug('Old value = %d', data[field2])
+            qset = set(quantities)
+            rlset = set(restock_levels)
+            rlset.remove(data[field2])
+            data[field2] = random.choice(list(rlset))
+            logging.debug('New value = %d', data[field2])
+        elif field2 == 'active':
+            logging.debug('Old value = %r', data[field2])
+            data[field2] = not data[field2]
+            logging.debug('New value = %r', data[field2])
+
+        # Make call to update record
+        response = self.client.put("{}/{}/{}".format(BASE_URL, data['product_id'], data['condition']), json=data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        updated_record = response.get_json()
+        self.assertEqual(updated_record[field1], data[field1])
+        self.assertEqual(updated_record[field2], data[field2])
+           
```

Comment:
This looks great ! Can we even check for a wrongly updated value ? As in if the user updates a field to a negative number . Then I would thrown an error and abort.

Reply:
Added that functionality